### PR TITLE
Adding endpoint to verify email, adding email verification functionality

### PR DESCRIPTION
**Pull request by polipopoliko**

### CODE REVIEW

File: api/email/verify_email_token.go [hunk 1]
```diff
@@ -0,0 +1,67 @@
...
+
+		err = db.QueryRow("SELECT et_id, token, email FROM email_token WHERE token=$1 AND email=$2", et.Token, et.Email).Scan(&et.Et_id, &et.Token, &et.Email)
+		if err != nil {
+			helpers.HTTPError(w, http.StatusBadRequest, "Unable to verify email token: "+err.Error())
+			return
+		}
+
+		err = UpdateCustomerVerificationStatus(et.Email, db)
+
+		if err != nil {
```
Comment (yulyulyharuka) [inline api/email/verify_email_token.go:47]: return supaya kalo error ga lanjut

File: api/email/verify_email_token.go [hunk 2]
```diff
@@ -0,0 +1,67 @@
...
+			helpers.HTTPError(w, http.StatusBadRequest, err.Error())
+		}
+
+		err = DeleteVerifiedEmailToken(et.Et_id, db)
+		if err != nil {
+			helpers.HTTPError(w, http.StatusBadRequest, "Unable to delete verified email: "+err.Error())
+			return
+		}
+		fmt.Fprintf(w, "Email %v has been succesfully verified at %v\n", et.Email, time.Now())
```
Comment (yulyulyharuka) [inline api/email/verify_email_token.go]: return json {
"email" : "email@example.com",
"status": "verified"
}

**Review by yulyulyharuka — CHANGES_REQUESTED** [review]

File: api/email/verify_email_token.go [hunk 3]
```diff
@@ -0,0 +1,78 @@
...
+		err = json.Unmarshal(requestBody, &et)
+		if err != nil {
+			helpers.HTTPError(w, http.StatusBadRequest, err.Error())
+			return
+		}
+
+		if err != nil {
+			helpers.HTTPError(w, http.StatusBadRequest, "Unable to verify email token: "+err.Error())
+			return
+		}
```
Comment (vicimilenia) [inline api/email/verify_email_token.go]: this part will never run because you already return in line 39

**Review by vicimilenia — CHANGES_REQUESTED** [review]

**Review by david1312 — APPROVED** [review]
good to go

**Review by yulyulyharuka — APPROVED** [review]

**Pull request merged**